examples/button.rs

```diff
@@ -0,0 +1,591 @@
+        default: Color::DARK_GRAY,
+    },
+    background_disabled: ButtonColor {
+        pressed: Color::rgb(0.16, 0.16, 0.16),
+        released: Color::rgb(0.16, 0.16, 0.16),
+        hovered: Color::rgb(0.16, 0.16, 0.16),
+        default: Color::rgb(0.15, 0.15, 0.15),
+    },
+    foreground_enabled: ButtonColor {
+        pressed: Color::BLUE,
+        released: Color::rgb(0.905, 0.921, 0.941),
+        hovered: Color::rgb(0.905, 0.921, 0.941),
+        default: Color::rgb(0.905, 0.921, 0.941),
+    },
+    foreground_disabled: ButtonColor {
+        pressed: Color::rgb(0.35, 0.35, 0.35),
+        released: Color::rgb(0.35, 0.35, 0.35),
+        hovered: Color::rgb(0.35, 0.35, 0.35),
+        default: Color::rgb(0.35, 0.35, 0.35),
+    },
+};
+
+/// Camera
+fn setup_camera(mut cmd: Commands) {
+    cmd.spawn_bundle(Camera2dBundle::default());
+}
+
+// Button can be interacted with
+// It has some content
+// That content is either an icon, some text or both
+fn setup_page(mut cmd: Commands, asset_server: Res<AssetServer>) {
+    // root node
+    cmd.spawn_bundle(NodeBundle {
+            style: Style {
+                size: Size::new(Val::Percent(100.0), Val::Percent(100.0)),
+                justify_content: JustifyContent::Center,
+                flex_direction: FlexDirection::Row,
+                ..default()
+            },
+            color: COLOR_BACKGROUND.into(),
+            ..default()
+        }).with_children(| root| {
+            // Content container
+            root.spawn_bundle(NodeBundle {
+                style: Style {
+                    size: Size::new(Val::Auto, Val::Percent(100.0)),
+                    min_size: Size::new(Val::Px(800.0), Val::Auto),
+                    padding: UiRect::all(Val::Px(30.0)),
+                    flex_direction: FlexDirection::ColumnReverse,
+                    justify_content: JustifyContent::FlexStart,
+                    align_items: AlignItems::FlexStart,
+                    ..default()
+                },
+                color: COLOR_BACKGROUND.into(),
+                ..default()
+            }).with_children(| content | {
+                // Title text
+                content.spawn_bundle(TextBundle::from_section(
+                    "Buttons", 
+                    TextStyle {
+                        font: asset_server.load(H1_FONT).into(),
+                        font_size: H1_FONT_SIZE,
+                        color: COLOR_TEXT,
+                    }
+                ).with_style(Style {
+                     margin: UiRect::new(Val::Undefined, Val::Undefined, Val::Px(15.0), Val::Undefined),
+                    ..default()
+                }));
+                // Paragraph
+                content.spawn_bundle(TextBundle::from_section(
+                    "Buttons are used to trigger actions. They can be clicked and hovered.",
+                    TextStyle {
+                        font: asset_server.load(TEXT_FONT).into(),
```

Comment:
You could reduce some boilerplate by making a variable for the font handle that you can reuse later. I also think the `into()` isn't needed here.

Something like
```rust
let text_font = asset_server.load(TEXT_FONT);

// -- snip --

font: text_font,
```

Reply:
(You might have to clone the handles though when you use them)